src/components/Modal.js

```diff
@@ -0,0 +1,43 @@
+export function Modal({ isOpen, onClose, title, children, allowClose = true }) {
+  if (!isOpen) return null;
+
+  const handleBackdropClick = (e) => {
+    // Cerrar solo si se clicka en el backdrop (y no en el contenido del modal)
+    if (allowClose && e.target === e.currentTarget) {
+      onClose();
+    }
+  };
+
+  // El contenido del modal en sí no debería causar un backdrop click
+  const handleModalClick = (e) => {
```

Comment:
Con tu otro check the target === currentTarget eso no es necesario, pero tampoco esta mal!
Solo creo que no ayuda lo suficiente para tenerlo porque se lanza ese con cualquier click en cualquier elemento hijo